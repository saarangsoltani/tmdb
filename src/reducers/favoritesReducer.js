import {
  ADD_FAVORITE_ITEM,
  REMOVE_FAVORITE_ITEM
} from "../actions/types";
import { loadState } from "../localStorage";

let initialState = loadState("favorites") || [];

export const favoritesReducer = (state = initialState, action) => {
  switch (action.type) {
    case ADD_FAVORITE_ITEM:
      return insertArrayItem(state, action.payload);
    case REMOVE_FAVORITE_ITEM:
      return removeArrayItem(state, action.payload);
    default:
      return state;
  }
};

function insertArrayItem(array, item) {
  let newArray = array.slice();
  newArray.push(item);
  return newArray;
}

function removeArrayItem(array, itemId) {
  let newArray = array.slice();
  var index = array.findIndex(fav => fav.id === itemId);
  newArray.splice(index, 1);
  return newArray;
}
