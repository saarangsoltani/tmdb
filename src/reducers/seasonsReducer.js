import { SET_SELECTED_SEASON } from "../actions/types";
let initialState = { selectedSeason: {episodes: []} };

export const seasonsReducer = (state = initialState, action) => {
  switch (action.type) {
    case SET_SELECTED_SEASON:
      return Object.assign({}, state, { selectedSeason: action.payload });
    default:
      return state;
  }
};
