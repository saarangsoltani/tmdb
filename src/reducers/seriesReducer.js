import {
  SET_DISCOVERED_SERIES,
} from "../actions/types";

let initialState = {
  selectedSeries: { seasons: [] },
  discovered: { page: 1, total_results: 0, total_pages: 0, results: [] }
};

export const seriesReducer = (state = initialState, action) => {
  switch (action.type) {
    case SET_DISCOVERED_SERIES:
      return Object.assign({}, state, { discovered: action.payload });
    default:
      return state;
  }
};
