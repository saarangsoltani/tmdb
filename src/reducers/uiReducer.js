import { API_START, API_END } from "../actions/types";

let initialState = {
  isLoading: false
};

export const uiReducer = (state = initialState, action) => {
  switch (action.type) {
    case API_START:
      return Object.assign({}, state, { isLoading: true });
    case API_END:
      return Object.assign({}, state, { isLoading: false });
    default:
      return state;
  }
};
