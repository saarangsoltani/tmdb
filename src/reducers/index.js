import { combineReducers } from "redux";
import { connectRouter } from "connected-react-router";
import { seriesReducer } from "./seriesReducer";
import { genresReducer } from "./genresReducer";
import { uiReducer } from "./uiReducer";

export default history =>
  combineReducers({
    router: connectRouter(history),
    series: seriesReducer,
    genres: genresReducer,
    ui: uiReducer
  });
