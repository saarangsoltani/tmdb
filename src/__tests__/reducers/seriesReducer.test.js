import { seriesReducer } from "../../reducers/seriesReducer";
import {
  SET_DISCOVERED_SERIES,
  SET_SELECTED_SERIES
} from "../../actions/types";

let initialState = seriesReducer(undefined, { type: "INIT" });

describe("reducers/series", () => {
  const discovered = {
    page: 1,
    total_results: 1,
    total_pages: 1,
    results: [{ name: "Friends" }]
  };

  it("should set discovered series to series list", () => {
    expect(
      seriesReducer(initialState, {
        type: SET_DISCOVERED_SERIES,
        payload: discovered
      })
    ).toMatchSnapshot();
  });
});
