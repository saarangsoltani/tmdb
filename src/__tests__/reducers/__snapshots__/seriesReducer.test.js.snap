// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers/series should set discovered series to series list 1`] = `
Object {
  "discovered": Object {
    "page": 1,
    "results": Array [
      Object {
        "name": "Friends",
      },
    ],
    "total_pages": 1,
    "total_results": 1,
  },
  "selectedSeries": Object {
    "seasons": Array [],
  },
}
`;
